# AlphaFold Protein Structure Database

### By Serena Chuang, Jessica Wang, Annapurna Saladi (Group 36)

* [Protein Structure](#protein-structure)
* [What is AlphaFold?](#what-is-alphafold)
* [How to use?](#how-to-use)
* [Applications](#applications)
* [Ahicevements and Limitations](#achievements-and-limitations)
* [Sources](#sources)

---

## Protein Structure

---

## What is AlphaFold?

---

## How to use?

There are two ways to use AlphaFold. First, there is the [AlphaFold Protein Structure Database](https://alphafold.ebi.ac.uk/), which provides free accessible protein structure predictions. This database contains over 200 million entries, which covers a substantial amount of Uniprot, the database that contains protein sequences and annotations. 

![AlphaFold Database Image](https://github.com/serrachow/beng183images/blob/main/alphafold_db.png)

For example, let's use `PTBP1` as an example gene. We can type in `PTBP1` into the search bar, and select the first option, which shows `Polypyrimidine tract-binding protein 1`. 

![AlphaFold Database Example](https://github.com/serrachow/beng183images/blob/main/alphafold_db_example.png)

A list of what the AlphaFold Protein Structure Database offers:

* `PDB file`
* `mmCIF file`
* `Predicted aligned error`
* `Structure viewer`
* `AlphaMissense Pathogenicity Heatmap`

---

## Applications

---

## Achievements and Limitations

---

## Sources

---